### ISSUE TITLE
On running with Singularity / full user mode

### ISSUE THREAD

**Issue by msimonin** [google/android-emulator-container-scripts#189]
I need to run the generated docker images (the ones built with `emu-docker create`) with singularity [[1]]

[1]: https://sylabs.io/guides/3.6/user-guide/introduction.html

First thing: this is working quite well ! 

But since singularity doesn't allow port mapping when calling in user mode (/I may be wrong/), port collisions will occur when starting a second container in the same machine. 
It seems that this can be fixed by changing the ports in the `launch_emulator.sh` (console, debugger, grpc and socat ports) so I'm wondering if you'd accept to allow those ports to be parameterized by an environment variable (like ADB_KEY in the same file). We could default to the current values if the variables are unset.

**Comment by pokowaka** [comment]
Yes absolutely! As long as the defaults work for everyone this should be fine.

**Comment by jpcottin** [comment]
Singularity's user-mode restriction on port mapping is a Singularity-side constraint rather than something the container scripts can work around — the emulator's adb / gRPC / console all need network reachability to be useful from outside the container. If you can run Singularity with `--net` (network namespace) or use Singularity's instance / fakeroot modes, port mapping becomes available.

Closing as a Singularity platform limitation. If a specific change inside the image would make Singularity user-mode workable, please open a fresh issue with the exact constraint.
